core/chains/evm/gas/models.go

```diff
@@ -271,12 +271,25 @@ func (e *evmFeeEstimator) GetFee(ctx context.Context, calldata []byte, feeLimit
 		fee.DynamicTipCap = dynamicFee.TipCap
 		return
 	}
-
 	// get legacy fee
 	fee.Legacy, chainSpecificFeeLimit, err = e.EvmEstimator.GetLegacyGas(ctx, calldata, feeLimit, maxFeePrice, opts...)
 	if err != nil {
 		return
 	}
+
+	if e.geCfg.MaxGasCostWei() != nil {
+		var totalGasCost uint64
+		totalGasCost, err = commonfee.ApplyMultiplier(fee.Legacy.ToInt().Uint64(), e.geCfg.LimitMultiplier())
+		if err != nil {
+			return
+		}
+		maxGasCost := e.geCfg.MaxGasCostWei().ToInt().Uint64()
+		if totalGasCost > maxGasCost {
+			err = fmt.Errorf("estimated gas cost %d exceeds maximum allowed gas cost %d", totalGasCost, maxGasCost)
+			return
+		}
+	}
+
 	chainSpecificFeeLimit, err = commonfee.ApplyMultiplier(chainSpecificFeeLimit, e.geCfg.LimitMultiplier())
```

Comment:
Should this line stay paired with the get on 275? (would have to add an error check since it won't be last anymore)

Reply:
will take a look into it

Reply:
Yes, this calculation should be done before the "if e.geCfg.CostMax() != nil {" line, and then the checks inside for CostMax() need not handle the Multiplier.

Reply:
fixed. PTAL